.github/workflows/go-ci.yml

```diff
@@ -51,7 +51,7 @@ jobs:
         cache-dependency-path: go.sum
     - name: Install staticcheck
       if: matrix.go == '1.23.6'
-      run: go install honnef.co/go/tools/cmd/staticcheck@v0.6.0
+      run: GOTOOLCHAIN='go1.24.2' go install honnef.co/go/tools/cmd/staticcheck@v0.6.0
```

Comment:
is it better to just change L42 to use the new go version?

Reply:
I want to make sure we're still supporting N-1 (i.e. 1.23 AND 1.24) so I don't want to bump that yet. But we just need to make sure that staticcheck is being compiled with 1.24